cmd/agent/app/config.go

```diff
@@ -212,14 +211,18 @@ func (b *Builder) CreateAgent(mFactory metrics.Factory, logger *zap.Logger) (*Ag
 		b.CollectorServiceName = defaultCollectorServiceName
 	}
 
+	// Use static collectors if specified.
+	if len(b.CollectorHostPorts) != 0 {
+		d := discovery.FixedDiscoverer(b.CollectorHostPorts)
+		notifier := &discovery.Dispatcher{}
+		b = b.WithDiscoverer(d).WithDiscoveryNotifier(notifier)
+	}
+
 	discoveryMgr, err := b.enableDiscovery(b.channel, logger)
 	if err != nil {
 		return nil, errors.Wrap(err, "cannot enable service discovery")
 	}
 	var clientOpts *tchannelThrift.ClientOptions
-	if discoveryMgr == nil && b.CollectorHostPort != "" {
-		clientOpts = &tchannelThrift.ClientOptions{HostPort: b.CollectorHostPort}
```

Comment:
@yurishkuro we don't need clientOptions when using the real discovery system because internally we use a different version of tchannel that binds to our internal discovery system. Here, we need to pass in the static port or else tchannel can't find collector.